bin/reorder-keys

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+if [ ! -d exercises ] ; then
+    echo "\"exercises\" directory not found, make sure you're running from the repository root" >&2
+    exit 1
+fi
+
+for canonical_data_file in exercises/*/canonical-data.json ; do
+    jq 'walk(if (type == "object") and has("uuid") then . as $obj | reduce ["uuid", "reimplements", "description", "comments", "scenarios", "property", "input", "expected"][] as $key ({}; . + (if $obj | has($key) then { ($key): $obj[$key] } else {} end)) else . end)' $canonical_data_file > tmp && mv tmp $canonical_data_file    
```

Comment:
```suggestion
    jq 'walk(if (type == "object") and has("uuid") then . as $obj | reduce ["uuid", "reimplements", "description", "comments", "scenarios", "property", "input", "expected"][] as $key ({}; . + (if $obj | has($key) then { ($key): $obj[$key] } else {} end)) else . end)' "$canonical_data_file" > tmp && mv tmp "$canonical_data_file"
```

Removes trailing whitespace and addresses `shellcheck` output.

> [SC2086](https://github.com/koalaman/shellcheck/wiki/Sc2086) (info): Double quote to prevent globbing and word splitting.

(I didn't review this PR properly - I just ran `shellcheck`).